src/detect-mqtt-subscribe-topic.c

```diff
@@ -78,8 +78,8 @@ static InspectionBuffer *MQTTSubscribeTopicGetData(DetectEngineThreadCtx *det_ct
 
     const uint8_t *data;
     uint32_t data_len;
-    if (rs_mqtt_tx_get_subscribe_topic(cbdata->txv, (uint16_t)cbdata->local_id,
-                &data, &data_len) == 0) {
+    if (rs_mqtt_tx_get_subscribe_topic(cbdata->txv, (uint32_t)cbdata->local_id, &data, &data_len) ==
```

Comment:
this uint32_t cast looks suspicious. Why not update the `local_id` var to actually be `uint32_t` in `MQTTSubscribeTopicGetDataArgs`?

Reply:
Good point. I guess that's what you get for starting off other code, you don't see the control you actually have. Of course that's the cleanest way to do it.

Reply:
Done for next PR